$OpenBSD: patch-panels_info_cc-info-overview-panel_c,v 1.4 2019/05/09 21:04:20 ajacoutot Exp $

No udisks2 on OpenBSD (XXX add HAVE_UDISKS2 to meson_options.txt)

Implement minimal support for get_os_name() on OpenBSD

Index: panels/info/cc-info-overview-panel.c
--- panels/info/cc-info-overview-panel.c.orig
+++ panels/info/cc-info-overview-panel.c
@@ -36,10 +36,16 @@
 #include <glibtop/mountlist.h>
 #include <glibtop/mem.h>
 #include <glibtop/sysinfo.h>
+#ifdef HAVE_UDISKS2
 #include <udisks/udisks.h>
+#endif
 
 #include <gdk/gdk.h>
 
+#ifdef __OpenBSD__
+#include <sys/utsname.h>
+#endif
+
 #ifdef GDK_WINDOWING_WAYLAND
 #include <gdk/gdkwayland.h>
 #endif
@@ -80,7 +86,11 @@ typedef struct
 
   GCancellable   *cancellable;
 
+#if HAVE_UDISKS2
   UDisksClient   *client;
+#else
+  void           *client;
+#endif
 
   GraphicsData   *graphics_data;
 } CcInfoOverviewPanelPrivate;
@@ -429,6 +439,7 @@ get_os_info (void)
 static char *
 get_os_name (void)
 {
+#ifndef __OpenBSD__
   GHashTable *os_info;
   gchar *name, *version_id, *pretty_name, *build_id;
   gchar *result = NULL;
@@ -467,6 +478,16 @@ get_os_name (void)
   g_clear_pointer (&os_info, g_hash_table_destroy);
 
   return result;
+#else
+  gchar *result = NULL;
+  struct utsname u;
+  if (uname (&u) == -1)
+    result = g_strdup_printf (_("Unknown"));
+  else
+    result = g_strdup_printf ("%s %s %s", u.sysname, u.release, u.version);
+
+  return result;
+#endif
 }
 
 static char *
@@ -498,6 +519,7 @@ get_primary_disc_info (CcInfoOverviewPanel *self)
       return;
     }
 
+#ifdef HAVE_UDISKS2
   manager = udisks_client_get_object_manager (priv->client);
   objects = g_dbus_object_manager_get_objects (manager);
 
@@ -526,6 +548,7 @@ get_primary_disc_info (CcInfoOverviewPanel *self)
     {
       gtk_label_set_text (GTK_LABEL (priv->disk_label), _("Unknown"));
     }
+#endif
 }
 
 static char *
@@ -842,7 +865,9 @@ cc_info_overview_panel_init (CcInfoOverviewPanel *self
   else
     gtk_widget_destroy (priv->updates_button);
 
+#if HAVE_UDISKS2
   priv->client = udisks_client_new_sync (NULL, &error);
+#endif
 
   if (error != NULL)
       g_warning ("Unable to get UDisks client: %s. Disk information will not be available.",
